flink-rpc/flink-rpc-akka/src/main/java/org/apache/flink/runtime/rpc/pekko/PekkoRpcServiceUtils.java

```diff
@@ -294,15 +290,20 @@ public PekkoRpcServiceBuilder withCustomConfig(final Config customConfig) {
 
         @Override
         public PekkoRpcServiceBuilder withBindAddress(final String bindAddress) {
-            this.bindAddress = Preconditions.checkNotNull(bindAddress);
+            if (bindAddress != null) {
```

Comment:
why change you changed this - prior to the change we would get a null pointer for null so we would end up in error processing

Reply:
In fact, we must ensure `bindAddress` is not null. All the code paths show that check the value like below.
```
        if (bindAddress != null) {
            rpcServiceBuilder.withBindAddress(bindAddress);
        }
```
So the `Preconditions.checkNotNull` is meaningless.

Reply:
Yes. we must provide the bindAddress for `ActorSystem`.

Reply:
@beliefer sorry if I am not being clear.
1) if bindAddress is never null coming into this method, then we do not need the if or the Preconditions.checkNotNull . In this case we can remove the if and the Preconditions.checkNotNull
2) if bindAddress can be null coming into this method then Preconditions.checkNotNull will throw a null pointer exception. So the current change changes behaviour as the null pointer will not be thrown.